diff --git i/arch/riscv/dts/sun20i-d1-mangopi-mq-pro.dts w/arch/riscv/dts/sun20i-d1-mangopi-mq-pro.dts
index e2bb6bc16c1..60129bb53bf 100644
--- i/arch/riscv/dts/sun20i-d1-mangopi-mq-pro.dts
+++ w/arch/riscv/dts/sun20i-d1-mangopi-mq-pro.dts
@@ -6,20 +6,36 @@
 
 /dts-v1/;
 
 #include "sun20i-d1.dtsi"
 #include "sun20i-common-regulators.dtsi"
 
 / {
 	model = "MangoPi MQ Pro";
 	compatible = "widora,mangopi-mq-pro", "allwinner,sun20i-d1";
 
+	memory@40000000 {
+		device_type = "memory";
+		reg = <0x40000000 0x20000000>;
+	};
+
+	reserved-memory {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges;
+
+		fel-esp@48000000 {
+			reg = <0x48000000 0x00800000>;
+			no-map;
+		};
+	};
+
 	aliases {
 		ethernet0 = &rtl8723ds;
 		serial0 = &uart0;
 	};
 
 	chosen {
 		stdout-path = "serial0:115200n8";
 	};
 
 	leds {
